src/parser/parse_redirection.c

```diff
@@ -1,45 +1,100 @@
 #include <commands/commands.h>
 #include <parser/command_table.h>
+#include <executor/executor_utils.h>
 #include <stdlib.h>
 #include <../libft/libft.h>
 #include <parser/parser.h>
+#include <parser/parse_redirection.h>
 #include <stdio.h>
+#include <fcntl.h>
 
-void	init_files(t_files *files)
+/*
+	Assumes that string has been checked for quotes
+*/
+int	get_file_name_and_length(char *buffer, char *input)
 {
-	files->in = NULL;
-	files->out = NULL;
+	const int	file_name_len = get_arg_len(input, SPECIALS);
+	int			len_to_cpy;
+	char		*cursor;
+
+	ft_bzero(&buffer[0], BUFSIZ);
+	cursor = input;
+	if (*cursor == DOUBLE_QUOTES)
+	{
+		++(cursor);
+		len_to_cpy = file_name_len - 2;
+	}
+	else
+		len_to_cpy = file_name_len;
+	ft_strlcpy(&buffer[0], cursor, len_to_cpy + 1);
+	return (file_name_len);
 }
 
 /*
-	Points to files name and consumes that part of the user input string
+	Might still split this into diff functions
 */
-void	get_in_out_file(const char **input_ptr, const char id, t_files *files)
+void	open_in_mode(const char *file, t_files *files, int mode_id)
 {
-	++(*input_ptr);
-	skip_spaces(input_ptr);
-	if (id == LEFT_ANGLE)
-		files->in = *input_ptr;
-	else if (id == RIGHT_ANGLE)
-		files->out = *input_ptr;
-	*input_ptr += get_cmd_len(*input_ptr);
+	if (mode_id == LEFT_ANGLE)
+	{
+		if (files->in > 0)
+			close(files->in);
+		files->in = open(file, O_RDONLY, 0644);
```

Comment:
I think we can make a enum/define for the file modes.
I think it makes more sense to check if `files->in == INVALID` first, then we check files->in > STDIN`
same for files->out, and probably split like get_in_file and get_out_file.

Reply:
As entioned on slack the first check is to see if there are any open fds already, changed it so that it's a bit more clear

Reply:
Also split into in and outfile functions